eng/Get-ServiceModules.ps1

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env pwsh
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+    .Synopsis
+    Gets the modules that are built from the given service directory and SDK type.
+
+    .Parameter ServiceDirectory
+    Name of the service being built. For example, "eventhubs".
+    .Parameter SdkType
+    Type of SDK to find modules for.
+
+    .Example
+    .\eng\Get-ServiceModules "eventhubs" "client"
+    Will output "azure-messaging-eventhubs,azure-messaging-eventhubs-checkpointstore-blob" because
+    those are modules that are built by the 'client' SDK type.
+
+    .Example
+    .\eng\Get-ServiceModules "eventhubs" "data"
+    Will output "microsoft-azure-eventhubs,microsoft-azure-eventhubs-eph,microsoft-azure-eventhubs-extensions"
+    because those are modules that are built by the 'data' SDK type.
+#>
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $true, Position = 0)]
+    [ValidateNotNullOrEmpty()]
+    [string]$ServiceDirectory,
+
+    [Parameter(Mandatory = $true, Position = 1)]
+    [ValidateSet("data", "client")]
+    [string]$SdkType
+)
+
+# By default stop for any error.
+if (!$PSBoundParameters.ContainsKey("ErrorAction")) {
+    $ErrorActionPreference = "Stop"
+}
+
+$root = Join-Path "$PSScriptRoot/../sdk" $ServiceDirectory | Resolve-Path
+
+switch($SdkType) {
```

Comment:
@conniey I'm going to recommend that you not use the directory names to filter but instead look at the pom.xml files under that area subdirectory and use the parent entry to filter out which is which.
Take a look at what I'm doing for the dependency graph generation which only looks at client but you can see the logic. Here's the [loop](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/DependencyGraph/gen-dep-graph-data.ps1#L103). Here is where I'm [checking the parent](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/DependencyGraph/gen-dep-graph-data.ps1#L119). What you'd do set the ValidParent to azure-client-sdk-parent or azure-data-sdk-parent and key off of that. I've found that to be more accurate and it has the added benefit of not needing to be changed if we ever have to manipulate 'too long' directory names.